### ISSUE TITLE
[BR]: Python 3.11: re.error: global flags not at the start of the expression at position 13

### ISSUE THREAD

**Issue by hobbes1069** [fail2ban/fail2ban#3314]
### Environment:

- Fail2Ban version  : 0.11.2
- OS, including release name/version : Fedora Linux Rawhide
- [x] Fail2Ban installed via OS/distribution mechanisms
- [ ] You have not applied any additional foreign patches to the codebase
- [x] Some customizations were done to the configuration (provide details below is so)

### The issue:

As usual several tests break with updated versions of Python

https://bugzilla.redhat.com/show_bug.cgi?id=2074533

**Comment by hobbes1069** [comment]
Yeah, kind of a dupe but that one was closed. In either case thanks for the PR. I'll pull it as a patch and report any issues.

**Comment by sebres** [comment]
Hmm... what is ready? 😄 I've a lot things to complete, but I'm unsure I'd be accomplished soon... 
So since the 0.11 branch is stable, let's not retard the release of "fixed" stuff.
I'll see what I can pick up to the release right now and would then release ASAP.

**Comment by hobbes1069** [comment]
Well no rush now. I had a minute while waiting on a work response and reviewed the patch failures of the PR. 3 of them were workflow files so not needed and one fixed a regular expression that doesn't appear in 0.11.2. I did a `quilt refresh` and tried a build and had no issues.

**Comment by larsen0815** [comment]
I just came across the problem "global flags not at the start of the expression at position" after having updated my server to Debian Bookworm. The problematic filter contained `(?i)phpmyadmin`. Workaround was to change this to `(?i:phpmyadmin)`.

Afaiu, this problem shouldn't exist anymore as the aforementioned fix was released with version 1.0.2 at the end of 2022.
Any idea what's going on?

````
qatar:~# python3 -V
Python 3.11.2

qatar:~# dpkg -l fail2ban
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version                                     Architecture Description
+++-==============-===========================================-============-===================================================
ii  fail2ban       1:1.0.2-v.debian.12+p18.0.57.0+t231012.0645 all          ban hosts that cause multiple authentication errors

qatar:~# lsb
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 12 (bookworm)
Release:    12
Codename:   bookworm

Old version:
failregex = ^<HOST> -.*"GET /+(?i)phpmyadmin/(index.php|scripts/setup.php).* HTTP/.*" 404 .*$
````

**Comment by sebres** [comment]
Just for the record...

> Workaround was to change this to `(?i:phpmyadmin)`.

The workaround may work (and even justified), however the behavior will be different after that change.
Previously the flags `(?i)` was affecting the whole regex, so everything was case insensitive there, `(?i:...)` doing it for the group only, thus after change it can match fewer messages than before. To make it "backwards" compatible `(?i)` must be moved to the beging of RE (`failregex` or `prefregex`).
Just to ensure everybody really understands the issue.
